File: lib/backend/src/routes/java.ts
import { Elysia } from 'elysia';
import { JavaService } from '../services/JavaService';

export const javaRoutes = new Elysia({ prefix: '/api/java' })
  .get('/detect', async () => {
    const system = await JavaService.detectSystemJava();
    const bundled = JavaService.getBundledJavaPath();
    
    return {
      system,
      bundled,
      preferred: bundled || system
    };
  })
  .post('/resolve', async ({ body }) => {
    const { path } = body as { path: string };
    const resolved = await JavaService.resolveJavaPath(path);
    return { resolved };
  });

File: lib/backend/src/routes/java.py
from fastapi import APIRouter
from pydantic import BaseModel
from ..services.JavaService import JavaService

router = APIRouter(prefix="/api/java", tags=["java"])

class JavaResolveRequest(BaseModel):
    path: str

@router.get("/detect")
def detect_java():
    # Use existing JavaService logic
    try:
        java_path = JavaService.detect_system_java()
        version = JavaService.get_java_version(java_path) if java_path else None
        return {
            "path": java_path,
            "version": version,
            "valid": java_path is not None
        }
    except:
        return {}

@router.post("/resolve")
def resolve_java(req: JavaResolveRequest):
    try:
        resolved = JavaService.resolve_java_path(req.path)
        return {"resolved": resolved}
    except:
        return {"resolved": None}
